### PR TITLE
Guide

### PR DESCRIPTION
**Pull request by nikita-jpg**

### CODE REVIEW

File: src/components/OnboardingGuide/Guide/Guide.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+
+import {Button, Icon, Progress} from '@gravity-ui/uikit';
+import {SVGIconData} from '@gravity-ui/uikit/build/esm/components/Icon/types';
```
Comment (amje) [inline src/components/OnboardingGuide/Guide/Guide.tsx:4]: There is `IconData` type exported from root

File: src/components/OnboardingGuide/Guide/Guide.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+import './Guide.scss';
+
+const cnGuide = block('guide');
+
+export type GuideProps = {
+    title: string | React.ReactNode;
+    progressBarCurrentValue: number;
+    icon?: SVGIconData;
+
+    defaultExpand: boolean;
```
Comment (amje) [inline src/components/OnboardingGuide/Guide/Guide.tsx:18]: `defaultExpanded`

File: src/components/OnboardingGuide/Guide/Guide.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+    isLoading?: boolean;
```
Comment (amje) [inline src/components/OnboardingGuide/Guide/Guide.tsx:20]: `loading`

File: src/components/OnboardingGuide/Guide/Guide.tsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+    children?: React.ReactNode;
+
+    rollUpbuttonText?: string;
+    completeButtontext?: string;
+
+    onHeaderClick?: (newExpand: boolean) => void;
```
Comment (amje) [inline src/components/OnboardingGuide/Guide/Guide.tsx:27]: If it's a click handler then it should have event in arguments. If you need expanded state, change the name to `onExpand: (expanded: boolean) => void`

File: src/components/OnboardingGuide/GuideMenuItem/GuideMenuItem.scss [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+$space-2-xs: 4px;
+$space-xs: 8px;
+$space-s: 12px;
+$space-l: 20px;
+$header-height-s: 26px;
+
+#{$block} {
+    &__collapse {
+        padding: $space-s $header-height-s $space-s $header-height-s;
```
Comment (amje) [inline src/components/OnboardingGuide/GuideMenuItem/GuideMenuItem.scss:13]: Can be shorten to `$space-s $header-height-s`

File: src/components/OnboardingGuide/GuideMenuItem/GuideMenuItem.scss [hunk 6]
```diff
@@ -0,0 +1,150 @@
+@use '../../variables';
+
+$block: '.#{variables.$ns}guide-menu-item';
+
+$space-2-xs: 4px;
+$space-xs: 8px;
+$space-s: 12px;
+$space-l: 20px;
```
Comment (amje) [inline src/components/OnboardingGuide/GuideMenuItem/GuideMenuItem.scss:8]: There are spacing vars `--g-spacing-{1-10}` by step of 4

File: src/components/OnboardingGuide/GuideMenuItem/GuideMenuItem.scss [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+    &__divider {
+        border-block-start: 1px solid var(--yc-color-line-generic);
+    }
+
+    &__skeleton {
+        height: 26px;
+        display: flex;
+        align-items: center;
+        justify-content: space-between;
+        margin: $space-s $header-height-s $space-s $header-height-s;
```
Comment (amje) [inline src/components/OnboardingGuide/GuideMenuItem/GuideMenuItem.scss:114]: Also can be shorten

File: src/components/OnboardingGuide/GuideMenuItem/GuideMenuItem.scss [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+        justify-content: space-between;
+    }
+
+    &:not(&_hasArrow) .g-disclosure__trigger {
+        cursor: auto;
+    }
+
+    &:not(&_hasArrow) .yc-arrow-toggle {
+        visibility: hidden;
+    }
```
Comment (amje) [inline src/components/OnboardingGuide/GuideMenuItem/GuideMenuItem.scss:149]: Don't use selectors from other component's internals

File: src/components/OnboardingGuide/GuideMenuItem/GuideMenuItem.tsx [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+
+import {block} from '../../utils/cn';
+
+import './GuideMenuItem.scss';
+
+export type GuideMenuItemProps = {
+    title: string;
+    status?: 'finished' | 'inProgress';
+    children?: React.ReactNode;
+    buttons?: React.ReactNode[];
```
Comment (amje) [inline src/components/OnboardingGuide/GuideMenuItem/GuideMenuItem.tsx:14]: ReactNode is already includes iterable type, no need to add brackets

File: src/components/OnboardingGuide/GuideMenuItem/GuideMenuItem.tsx [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+    showDivider?: boolean;
+    hasArrow?: boolean;
+    isLoading?: boolean;
```
Comment (amje) [inline src/components/OnboardingGuide/GuideMenuItem/GuideMenuItem.tsx:17]: `loading`

File: src/components/OnboardingGuide/GuideMenuItem/GuideMenuItem.tsx [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+import {block} from '../../utils/cn';
+
+import './GuideMenuItem.scss';
+
+export type GuideMenuItemProps = {
+    title: string;
+    status?: 'finished' | 'inProgress';
+    children?: React.ReactNode;
+    buttons?: React.ReactNode[];
+    showDivider?: boolean;
```
Comment (amje) [inline src/components/OnboardingGuide/GuideMenuItem/GuideMenuItem.tsx:15]: `hasDivider`

File: src/components/OnboardingGuide/GuideMenuItem/GuideMenuItem.tsx [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+import {Clock} from '@gravity-ui/icons';
+import {Checkbox, Disclosure, Icon, Skeleton} from '@gravity-ui/uikit';
+
+import {block} from '../../utils/cn';
+
+import './GuideMenuItem.scss';
+
+export type GuideMenuItemProps = {
+    title: string;
+    status?: 'finished' | 'inProgress';
```
Comment (amje) [inline src/components/OnboardingGuide/GuideMenuItem/GuideMenuItem.tsx:12]: `completed` and `pending` better suits for status

File: src/components/OnboardingGuide/GuideMenuItem/frequentFunctions/GuideMenuItemImage/GuideMenuItemImage.scss [hunk 13]
```diff
@@ -0,0 +1,11 @@
+@use '../../../../variables.scss';
+$block: '.#{variables.$ns}guide-menu-item-image';
+
+#{$block} {
+    max-width: 242px;
+    max-height: 118px;
+
+    border: 1px solid var(--colors-line-generic, rgba(0, 0, 0, 0.1));
```
Comment (amje) [inline src/components/OnboardingGuide/GuideMenuItem/frequentFunctions/GuideMenuItemImage/GuideMenuItemImage.scss:8]: `rgba(0, 0, 0, 0.1)` would barely be visible in dark theme

File: src/components/OnboardingGuide/GuideMenuItem/frequentFunctions/GuideMenuItemImage/GuideMenuItemImage.tsx [hunk 14]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+
+import {block} from '../../../../utils/cn';
+
+import './GuideMenuItemImage.scss';
+
+const cnGuideMenuItemImage = block('guide-menu-item-image');
+
+export type GuideMenuItemImageProps = {
+    link: string;
```
Comment (amje) [inline src/components/OnboardingGuide/GuideMenuItem/frequentFunctions/GuideMenuItemImage/GuideMenuItemImage.tsx:10]: `src`

File: src/components/OnboardingGuide/i18n/en.json [hunk 15]
```diff
@@ -0,0 +1,4 @@
+{
+  "roll-up": "Roll up",
```
Comment (amje) [inline src/components/OnboardingGuide/i18n/en.json:2]: Maybe "Collapse" instead of "Roll up"?

File: src/components/OnboardingGuide/Guide/Guide.scss [hunk 16]
```diff
@@ -0,0 +1,112 @@
...
+
+#{$block} {
+    z-index: $layer-popup;
+    max-width: 280px;
+
+    background: none;
+    width: fit-content;
+
+    &__content {
+        box-shadow: 0 2px 8px 0 rgba(0, 0, 0, 0.04);
```
Comment (amje) [inline src/components/OnboardingGuide/Guide/Guide.scss:21]: `rgba(0, 0, 0, 0.04)` not be visible in dark theme, you should use variable from sfx group

File: src/components/OnboardingGuide/Guide/Guide.scss [hunk 17]
```diff
@@ -0,0 +1,112 @@
...
+
+        width: 280px;
+        display: flex;
+        flex-direction: column;
+        overflow: hidden;
+
+        box-sizing: border-box;
+        height: 62px;
+
+        background-color: var(--g-color-base-info-heavy);
```
Comment (amje) [inline src/components/OnboardingGuide/Guide/Guide.scss:31]: Why it is not brand color?

File: src/components/OnboardingGuide/Guide/Guide.tsx [hunk 18]
```diff
@@ -0,0 +1,140 @@
...
+    progressBarCurrentValue: number;
+    icon?: SVGIconData;
+
+    defaultExpand: boolean;
+
+    isLoading?: boolean;
+
+    children?: React.ReactNode;
+
+    rollUpbuttonText?: string;
```
Comment (amje) [inline src/components/OnboardingGuide/Guide/Guide.tsx:24]: "button" should starts with capital "B"

File: src/components/OnboardingGuide/Guide/Guide.tsx [hunk 19]
```diff
@@ -0,0 +1,140 @@
...
+    completeButtontext?: string;
+
+    onHeaderClick?: (newExpand: boolean) => void;
+    onCompleteClick: () => void;
+    onRollUpClick?: () => void;
```
Comment (amje) [inline src/components/OnboardingGuide/Guide/Guide.tsx:29]: events missing

File: src/components/OnboardingGuide/Guide/Guide.tsx [hunk 20]
```diff
@@ -0,0 +1,140 @@
...
+import {block} from '../../utils/cn';
+import i18n from '../i18n';
+
+import './Guide.scss';
+
+const cnGuide = block('guide');
+
+export type GuideProps = {
+    title: string | React.ReactNode;
+    progressBarCurrentValue: number;
```
Comment (amje) [inline src/components/OnboardingGuide/Guide/Guide.tsx:15]: Maybe shorten this to `progress`?

File: src/components/OnboardingGuide/Guide/Guide.tsx [hunk 21]
```diff
@@ -0,0 +1,140 @@
...
+
+import {block} from '../../utils/cn';
+import i18n from '../i18n';
+
+import './Guide.scss';
+
+const cnGuide = block('guide');
+
+export type GuideProps = {
+    title: string | React.ReactNode;
```
Comment (amje) [inline src/components/OnboardingGuide/Guide/Guide.tsx:14]: `string` included in `ReactNode`

File: src/components/OnboardingGuide/Guide/Guide.tsx [hunk 22]
```diff
@@ -0,0 +1,140 @@
...
+    icon?: SVGIconData;
+
+    defaultExpand: boolean;
+
+    isLoading?: boolean;
+
+    children?: React.ReactNode;
+
+    rollUpbuttonText?: string;
+    completeButtontext?: string;
```
Comment (amje) [inline src/components/OnboardingGuide/Guide/Guide.tsx:25]: "text" should starts with capital "T"

File: src/components/OnboardingGuide/Guide/Guide.tsx [hunk 23]
```diff
@@ -0,0 +1,140 @@
...
+        <div className={cnGuide({expanded: expand})}>
+            <div className={cnGuide('content', {expanded: expand})}>
+                <div
+                    className={cnGuide('header', {state: expand && 'expanded'})}
+                    onClick={onHeaderClickCallback}
+                    onKeyDown={onHeaderClickCallback}
+                    role="button"
+                    tabIndex={0}
+                >
+                    <button className={cnGuide('title')}>
```
Comment (amje) [inline src/components/OnboardingGuide/Guide/Guide.tsx:78]: You have button inside element with role="button", that's incorrect

File: src/components/OnboardingGuide/Guide/Guide.tsx [hunk 24]
```diff
@@ -0,0 +1,140 @@
...
+                            })}
+                        >
+                            {children}
+                        </div>
+                        <div className={cnGuide('buttons')}>
+                            <Button
+                                view="normal-contrast"
+                                size="m"
+                                onClick={onRollUpClickCallback}
+                                key={'rollUp'}
```
Comment (amje) [inline src/components/OnboardingGuide/Guide/Guide.tsx:121]: `key` not needed

File: src/components/OnboardingGuide/Guide/Guide.scss [hunk 25]
```diff
@@ -0,0 +1,112 @@
...
+        border-radius: 20px;
+
+        transition: height 300ms ease-out;
```
Comment (amje) [inline src/components/OnboardingGuide/Guide/Guide.scss:34]: Transition not work in stories (browser: Edge MacOS)

File: src/components/OnboardingGuide/Guide/Guide.tsx [hunk 26]
```diff
@@ -0,0 +1,140 @@
...
+        onCompleteClick();
+    };
+
+    return (
+        <div className={cnGuide({expanded: expand})}>
+            <div className={cnGuide('content', {expanded: expand})}>
+                <div
+                    className={cnGuide('header', {state: expand && 'expanded'})}
+                    onClick={onHeaderClickCallback}
+                    onKeyDown={onHeaderClickCallback}
```
Comment (amje) [inline src/components/OnboardingGuide/Guide/Guide.tsx:74]: It will trigger expand on every key press

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/OnboardingGuide/GuideMenuItem/GuideMenuItem.scss (hunk 8 above)
Comment (nikita-jpg) [reply src/components/OnboardingGuide/GuideMenuItem/GuideMenuItem.scss:149]: Can I do this? I have no other ideas how to stretch the Disclosure header to its full width

Comment (nikita-jpg) [reply src/components/OnboardingGuide/GuideMenuItem/GuideMenuItem.scss:149]: or can I write my own equivalent of Disclosure? It doesn't look difficult

Comment (nikita-jpg) [reply src/components/OnboardingGuide/GuideMenuItem/GuideMenuItem.scss:149]: the situation is similar with the checkbox :(

Comment (amje) [reply src/components/OnboardingGuide/GuideMenuItem/GuideMenuItem.scss:149]: You can render custom summary in disclojure: https://github.com/gravity-ui/uikit/blob/31800603eaf6eaaab4c9849b1af12e9e6a53b5f7/src/components/Disclosure/__stories__/Disclosure.stories.tsx#L63

File: src/components/OnboardingMenu/__stories__/OnboardingMenu.stories.tsx [hunk 27]
```diff
@@ -0,0 +1,196 @@
...
+    };
+
+    return (
+        <div style={{margin: '20px'}}>
+            <OnboardingMenu
+                {...args}
+                icon={GraduationCap}
+                onExpand={actionsOnboardingMenuHandlers}
+                onCompleteClick={actionsButtonComplete}
+                progress={0}
```
Comment (amje) [inline src/components/OnboardingMenu/__stories__/OnboardingMenu.stories.tsx:34]: This overrides args, so progress always 0

File: src/components/OnboardingMenu/components/OnboardingMenuItem/OnboardingMenuItem.scss [hunk 28]
```diff
@@ -0,0 +1,126 @@
...
+    }
+
+    &__title {
+        color: var(--g-color-text-complementary);
+
+        font-family: 'YS Text';
+        font-size: 15px;
+        font-style: normal;
+        font-weight: 400;
+        line-height: 20px;
```
Comment (amje) [inline src/components/OnboardingMenu/components/OnboardingMenuItem/OnboardingMenuItem.scss:121]: Use mixin here

File: src/components/OnboardingMenu/components/OnboardingMenuItem/OnboardingMenuItem.tsx [hunk 29]
```diff
@@ -0,0 +1,113 @@
...
+        );
+    }, []);
+
+    const summary = React.useMemo(() => {
+        return <span className={cnOnboardingMenuItem('title')}>{title}</span>;
+    }, []);
+
+    const header = React.useMemo(
+        () => (
+            <div className={cnOnboardingMenuItem('collapse-header')} role="button" tabIndex={0}>
```
Comment (amje) [inline src/components/OnboardingMenu/components/OnboardingMenuItem/OnboardingMenuItem.tsx:67]: Disclosure is already interactive element, we cant nest them, also tabIndex makes this element to be focusable, but no handlers are passed

File: src/components/OnboardingMenu/components/OnboardingMenuItem/OnboardingMenuItem.tsx [hunk 30]
```diff
@@ -0,0 +1,113 @@
...
+                className={cnOnboardingMenuItem('time-icon__disable')}
+            />
+        );
+    }, []);
+
+    const summary = React.useMemo(() => {
+        return <span className={cnOnboardingMenuItem('title')}>{title}</span>;
+    }, []);
+
+    const header = React.useMemo(
```
Comment (amje) [inline src/components/OnboardingMenu/components/OnboardingMenuItem/OnboardingMenuItem.tsx:65]: We don't need any memo here, just put markup as is

File: src/components/OnboardingMenu/components/OnboardingMenuItem/frequentFunctions/OnboardingMenuItemText/OnboardingMenuItemText.tsx [hunk 31]
```diff
@@ -0,0 +1,17 @@
...
+
+export type OnboardingMenuItemTextProps = {
+    text: string;
+};
+
+const OnboardingMenuItemText = ({text}: OnboardingMenuItemTextProps) => (
+    <span className={cnOnboardingMenuItemText()}>{text}</span>
+);
+
+export const addText = (text: string) => <OnboardingMenuItemText text={text} />;
```
Comment (amje) [inline src/components/OnboardingMenu/components/OnboardingMenuItem/frequentFunctions/OnboardingMenuItemText/OnboardingMenuItemText.tsx:17]: Why we need this helper? `OnboardingMenuItemText` is already act like a function of `text`

File: src/components/OnboardingMenu/components/OnboardingMenuItem/frequentFunctions/OnboardingMenuItemText/OnboardingMenutemText.scss [hunk 32]
```diff
@@ -0,0 +1,14 @@
...
+#{$block} {
+    overflow: hidden;
+    color: var(--g-color-text-complementary);
+    text-overflow: ellipsis;
+
+    font-family: 'YS Text';
+    font-size: 13px;
+    font-style: normal;
+    font-weight: 400;
+    line-height: 18px; /* 138.462% */
```
Comment (amje) [inline src/components/OnboardingMenu/components/OnboardingMenuItem/frequentFunctions/OnboardingMenuItemText/OnboardingMenutemText.scss:13]: See note above

File: src/components/OnboardingMenu/components/OnboardingMenuItem/frequentFunctions/OnboardingMenutemImage/OnboardingMenuItemImage.scss [hunk 33]
```diff
@@ -0,0 +1,11 @@
+@use '../../../../../variables.scss';
+$block: '.#{variables.$ns}guide-menu-item-image';
+
+#{$block} {
+    max-width: 242px;
+    max-height: 118px;
+
+    border: 1px solid var(--g-color-line-generic, rgba(0, 0, 0, 0.1));
```
Comment (amje) [inline src/components/OnboardingMenu/components/OnboardingMenuItem/frequentFunctions/OnboardingMenutemImage/OnboardingMenuItemImage.scss:8]: `--g-color-line-generic` is always declared, no need in default value

File: src/components/OnboardingMenu/components/OnboardingMenuItem/frequentFunctions/OnboardingMenutemImage/OnboardingMenuItemImage.tsx [hunk 34]
```diff
@@ -0,0 +1,20 @@
...
+export type OnboardingMenuItemImageProps = {
+    src: string;
+    alt: string;
+};
+
+export const OnboardingMenuItemImage = ({src, alt}: OnboardingMenuItemImageProps) => (
+    <img src={src} alt={alt} className={cnOnboardingMenuItemImage()}></img>
+);
+
+export const addImage = (src: string, alt: string) => (
```
Comment (amje) [inline src/components/OnboardingMenu/components/OnboardingMenuItem/frequentFunctions/OnboardingMenutemImage/OnboardingMenuItemImage.tsx:18]: Same note about helper as above

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/OnboardingMenu/OnboardingMenu.scss [hunk 35]
```diff
@@ -0,0 +1,121 @@
...
+        margin-block-end: var(--g-spacing-4);
+
+        margin-inline-start: 1px;
+
+        background-color: var(--g-color-base-float);
+
+        border-radius: var(--g-border-radius-xl);
+    }
+
+    &__lining::-webkit-scrollbar {
```
[the remark this answers is not in the document]
Comment (nikita-jpg) [reply src/components/OnboardingMenu/OnboardingMenu.scss:108]: Added an example with a lot of elements to storybook. 
I don't want to show them the scroll. Is this a bad idea?

File: src/components/OnboardingMenu/components/OnboardingMenuItem/frequentFunctions/OnboardingMenuItemText/OnboardingMenuItemText.tsx (hunk 31 above)
Comment (nikita-jpg) [reply src/components/OnboardingMenu/components/OnboardingMenuItem/frequentFunctions/OnboardingMenuItemText/OnboardingMenuItemText.tsx:17]: it seemed to me that it looks more convenient than using the component itself.

`addText('Test text')`
```
{[
    addImage(
        'https://storage.yandexcloud.net/uikit-storybook-assets/changelog-dialog-picture-1.png',
        '2019',
    ),
    addText('Test text')
]}
```

or

`<OnboardingMenuItemText text={'Test text'}/>`
```
{[
    addImage(
        'https://storage.yandexcloud.net/uikit-storybook-assets/changelog-dialog-picture-1.png',
        '2019',
    ),
    <OnboardingMenuItemText text={'Test text'}/>
]}
```

File: src/components/OnboardingMenu/components/OnboardingMenuItem/OnboardingMenuItem.scss [hunk 36]
```diff
@@ -0,0 +1,126 @@
...
+        margin-inline-end: var(--g-spacing-2);
+
+        align-self: flex-start;
+        display: flex;
+        align-items: center;
+        height: --g-spacing-5;
+    }
+
+    &__time-icon {
+        color: var(--g-color-base-brand);
```
[the remark this answers is not in the document]
Comment (nikita-jpg) [reply src/components/OnboardingMenu/components/OnboardingMenuItem/OnboardingMenuItem.scss:45]: This is the color of the icon, not the text

File: src/components/OnboardingMenu/components/OnboardingMenuItem/frequentFunctions/OnboardingMenuItemText/OnboardingMenuItemText.tsx (hunk 31 above)
Comment (amje) [reply src/components/OnboardingMenu/components/OnboardingMenuItem/frequentFunctions/OnboardingMenuItemText/OnboardingMenuItemText.tsx:17]: Feels unnatural to use React like that

File: src/components/OnboardingMenu/OnboardingMenu.scss (hunk 35 above)
[the remark this answers is not in the document]
Comment (amje) [reply src/components/OnboardingMenu/OnboardingMenu.scss:108]: Yes, hiding scroll is bad

File: src/components/OnboardingMenu/OnboardingMenu.scss [hunk 37]
```diff
@@ -0,0 +1,108 @@
...
+        color: var(--g-color-text-brand-contrast);
+        @include mixins.text-subheader-2();
+
+        overflow: hidden;
+        text-overflow: ellipsis;
+        text-wrap: nowrap;
+
+        &_expand {
+            @include mixins.text-subheader-3();
+        }
```
Comment (amje) [inline src/components/OnboardingMenu/OnboardingMenu.scss:44]: It's better to put color and font styles on the button, thus you don't need to duplicate them on text and icon elements

File: src/components/OnboardingMenu/OnboardingMenu.scss [hunk 38]
```diff
@@ -0,0 +1,121 @@
...
+    }
+
+    &__title {
+        display: flex;
+        align-items: center;
+        justify-content: center;
+
+        margin-block-end: var(--g-spacing-1);
+
+        color: var(--g-color-text-light-primary);
```
[the remark this answers is not in the document]
Comment (amje) [reply src/components/OnboardingMenu/OnboardingMenu.scss:54]: What is this line for?

File: src/components/OnboardingMenu/OnboardingMenu.tsx [hunk 39]
```diff
@@ -0,0 +1,129 @@
...
+    defaultExpanded: boolean;
+    children?: React.ReactNode;
+
+    collapseButtonText?: string;
+    completeButtonText?: string;
+    className?: string;
+
+    onExpand?: (expanded: boolean) => void;
+    onCompleteClick: () => void;
+    onRollUpClick?: (event: React.MouseEvent) => void;
```
Comment (amje) [inline src/components/OnboardingMenu/OnboardingMenu.tsx:27]:
```suggestion
    onCollapseClick?: (event: React.MouseEvent) => void;
```

File: src/components/OnboardingMenu/OnboardingMenu.tsx [hunk 40]
```diff
@@ -0,0 +1,129 @@
...
+
+    defaultExpanded: boolean;
+    children?: React.ReactNode;
+
+    collapseButtonText?: string;
+    completeButtonText?: string;
+    className?: string;
+
+    onExpand?: (expanded: boolean) => void;
+    onCompleteClick: () => void;
```
Comment (amje) [inline src/components/OnboardingMenu/OnboardingMenu.tsx:26]:
```suggestion
    onCompleteClick: (event: React.MouseEvent) => void;
```

File: src/components/OnboardingMenu/components/OnboardingMenuItem/OnboardingMenuItem.scss [hunk 41]
```diff
@@ -0,0 +1,102 @@
...
+        width: 100%;
+    }
+
+    &__icon {
+        margin-inline-end: var(--g-spacing-2);
+
+        align-self: flex-start;
+        display: flex;
+        align-items: center;
+        height: --g-spacing-5;
```
Comment (amje) [inline src/components/OnboardingMenu/components/OnboardingMenuItem/OnboardingMenuItem.scss:23]: Do you mean this?

```suggestion
        height: var(--g-spacing-5);
```

File: src/components/OnboardingMenu/components/OnboardingMenuItem/OnboardingMenuItem.scss (hunk 36 above)
[the remark this answers is not in the document]
Comment (amje) [reply src/components/OnboardingMenu/components/OnboardingMenuItem/OnboardingMenuItem.scss:27]: icon has text semantic not background, base-* colors are for the backgrounds

File: src/components/OnboardingMenu/components/OnboardingMenuItem/OnboardingMenuItem.tsx [hunk 42]
```diff
@@ -0,0 +1,106 @@
...
+    ) : (
+        summary
+    );
+
+    if (loading) {
+        return (
+            <React.Fragment>
+                <div className={cnOnboardingMenuItem('skeleton')}>
+                    <Skeleton
+                        className={cnOnboardingMenuItem('avatar')}
```
Comment (amje) [inline src/components/OnboardingMenu/components/OnboardingMenuItem/OnboardingMenuItem.tsx:90]:
```suggestion
                        className={cnOnboardingMenuItem('skeleton-avatar')}
```

File: src/components/OnboardingMenu/components/OnboardingMenuItem/OnboardingMenuItem.tsx [hunk 43]
```diff
@@ -0,0 +1,106 @@
...
+                        style={{width: `${ICON_SIZE}px`, height: `${ICON_SIZE}px`}}
+                    />
+                    <Skeleton className={cnOnboardingMenuItem('person')} />
```
Comment (amje) [inline src/components/OnboardingMenu/components/OnboardingMenuItem/OnboardingMenuItem.tsx:93]:
```suggestion
                    <Skeleton className={cnOnboardingMenuItem('skeleton-title')} />
```

File: src/components/OnboardingMenu/components/OnboardingMenuItem/OnboardingMenuItem.scss [hunk 44]
```diff
@@ -0,0 +1,102 @@
+@use '../../../variables.scss';
+@use '@gravity-ui/uikit/styles/mixins';
+
+$block: '.#{variables.$ns}guide-menu-item';
```
Comment (amje) [inline src/components/OnboardingMenu/components/OnboardingMenuItem/OnboardingMenuItem.scss:4]: Item should match the class of the parent

File: src/components/OnboardingMenu/components/OnboardingMenuItem/frequentFunctions/OnboardingMenuItemText/OnboardingMenuItemText.tsx [hunk 45]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
```
Comment (amje) [inline src/components/OnboardingMenu/components/OnboardingMenuItem/frequentFunctions/OnboardingMenuItemText/OnboardingMenuItemText.tsx:1]: Do we really need to extract `MenuItemText` and `MenuItemImage` to separate components?I don't think so, let's keep them in the `MenuItem` component

File: src/components/OnboardingMenu/OnboardingMenu.tsx [hunk 46]
```diff
@@ -0,0 +1,129 @@
...
+    onRollUpClick?: (event: React.MouseEvent) => void;
+};
+
+export const OnboardingMenu = ({
```
Comment (amje) [inline src/components/OnboardingMenu/OnboardingMenu.tsx:30]: You must also reexport all of the public components in the `src/components/OnboardingMenu/index.ts` file, and also add reexport to the `src/components/index.ts`

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/OnboardingMenu/OnboardingMenu.scss [hunk 47]
```diff
@@ -0,0 +1,108 @@
+@use '../variables.scss';
+@use '@gravity-ui/uikit/styles/mixins';
+
+$block: '.#{variables.$ns}guide';
```
[the remark this answers is not in the document]
Comment (nikita-jpg) [reply src/components/OnboardingMenu/OnboardingMenu.scss:4]: https://st.yandex-team.ru/UXRFC-255#65e6e0c02415f11fbb9199cc

File: src/components/OnboardingMenu/components/OnboardingMenuItem/frequentFunctions/OnboardingMenuItemText/OnboardingMenuItemText.tsx (hunk 45 above)
Comment (nikita-jpg) [reply src/components/OnboardingMenu/components/OnboardingMenuItem/frequentFunctions/OnboardingMenuItemText/OnboardingMenuItemText.tsx:1]: 1) What is the problem?
2) Different components look more convenient for `OnboardingMenuItem`

**Review by amje — COMMENTED** [review]
Let's use this naming:
```
OnboardingMenu.Item
OnboardingMenu.ItemText
OnboardingMenu.ItemImage
```

File: src/components/OnboardingMenu/OnboardingMenu.tsx [hunk 48]
```diff
@@ -128,3 +131,9 @@ export const OnboardingMenu = ({
         </div>
     );
 };
+
+export const OnboardingMenu = Object.assign(OnboardingMenuParent, {
+    Item: OnboardingMenuItem,
+    ItemText: OnboardingMenuItemText,
+    ItemImage: OnboardingMenuItemImage,
+});
```
Comment (amje) [inline src/components/OnboardingMenu/OnboardingMenu.tsx:139]: We should also add `OnboardingMenu.displayName = 'OnboardingMenu';`, otherwise it would be `OnboardingMenuParent `

**Review by amje — CHANGES_REQUESTED** [review]

**Review by amje — APPROVED** [review]
